### PR TITLE
[Admin] Add commands for reading entities

### PR DESCRIPTION
**Pull request by smnzhu**
* Add built-in admin command to list all commands
* Refactor read-blocks command
* Change the return type of `unittest.BlockFixture()` to pointer type
* Refactor helper functions for all storage-related commands
* Create read-results and read-seals commands

### CODE REVIEW

File: admin/commands/storage/read_blocks_test.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+	genesis := unittest.GenesisFixture()
+	blocks = append(blocks, genesis)
+	sealed := unittest.BlockWithParentFixture(genesis.Header)
+	blocks = append(blocks, sealed)
+	final := unittest.BlockWithParentFixture(sealed.Header)
+	blocks = append(blocks, final)
+	final = unittest.BlockWithParentFixture(final.Header)
+	blocks = append(blocks, final)
+	final = unittest.BlockWithParentFixture(final.Header)
+	blocks = append(blocks, final)
```
Comment (kc1116) [inline admin/commands/storage/read_blocks_test.go:67]: Question: Why do these 2 "final" blocks both have the same parent ?

File: admin/commands/storage/read_blocks_test.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	}))
+	assert.Error(suite.T(), suite.command.Validator(&admin.CommandRequest{
+		Data: 420,
+	}))
+	assert.Error(suite.T(), suite.command.Validator(&admin.CommandRequest{
+		Data: "foo",
+	}))
+	assert.Error(suite.T(), suite.command.Validator(&admin.CommandRequest{
+		Data: map[string]interface{}{
+			"glock": 123,
```
Comment (kc1116) [inline admin/commands/storage/read_blocks_test.go:129]:
```suggestion
			"block": 123,
```

File: admin/commands/storage/read_blocks.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	}
+
+	return convertToInterfaceList(result)
+}
+
+func (r *ReadBlocksCommand) Validator(req *admin.CommandRequest) error {
+	input, ok := req.Data.(map[string]interface{})
+	if !ok {
+		return errors.New("wrong input format: expected JSON")
+	}
```
Comment (kc1116) [inline admin/commands/storage/read_blocks.go:57]: Should return sentinel errors for invalid request data to be asserted in the test, something like the following

```
var (
    ErrValidatorReqDataFormat = fmt.Errorf("wrong input format: expected JSON"
)
```

```suggestion
	if !ok {
		return ErrValidatorReqDataFormat
	}
```

File: admin/commands/storage/read_blocks_test.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+		},
+	}))
+}
+
+func (suite *ReadBlocksSuite) TestValidateInvalidBlock() {
```
Comment (kc1116) [inline admin/commands/storage/read_blocks_test.go:134]: Followup to suggestion above, should we return sentinel errors for these invalid data inputs ?

File: admin/commands/storage/read_results.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+		resultID = result.PreviousResultID
+	}
+
+	return convertToInterfaceList(results)
+}
+
+func (r *ReadResultsCommand) Validator(req *admin.CommandRequest) error {
+	input, ok := req.Data.(map[string]interface{})
+	if !ok {
+		return errors.New("wrong input format: expected JSON")
```
Comment (kc1116) [inline admin/commands/storage/read_results.go:71]: we can move this and reuse this error message in all the Validator(req *admin.CommonRequest) funcs

File: admin/commands/storage/read_results.go [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	data := &readResultsRequest{}
+
+	if result, ok := input["result"]; ok {
+		errInvalidResultValue := fmt.Errorf("invalid value for \"result\": expected a result ID represented as a 64 character long hex string, but got: %v", result)
```
Comment (kc1116) [inline admin/commands/storage/read_results.go:77]: Same comment about errors from above

**Review by kc1116 — APPROVED** [review]
Small suggestions and a question

**Review by zhangchiqing — APPROVED** [review]
Nice work! 

I just tried, works great. 💯 

Ideally, there could be more entities to be queried. For instance, receipts, transactions, chunk data pack, etc. But we could add them in different PRs.

**Comment by smnzhu** [comment]
bors merge

**Comment by smnzhu** [comment]
bors merge

**Comment by smnzhu** [comment]
bors merge

**Comment by smnzhu** [comment]
bors merge

**Pull request merged**